homeassistant/components/withings/sensor.py

```diff
@@ -363,11 +365,31 @@ async def async_setup_entry(
     measurement_coordinator: WithingsMeasurementDataUpdateCoordinator = hass.data[
         DOMAIN
     ][entry.entry_id][MEASUREMENT_COORDINATOR]
+
+    current_measurement_types = set(measurement_coordinator.data.keys())
+
     entities: list[SensorEntity] = []
     entities.extend(
-        WithingsMeasurementSensor(measurement_coordinator, attribute)
-        for attribute in MEASUREMENT_SENSORS
+        WithingsMeasurementSensor(
+            measurement_coordinator, MEASUREMENT_SENSORS[measurement_type]
+        )
+        for measurement_type in measurement_coordinator.data
+        if measurement_type in MEASUREMENT_SENSORS
     )
+
+    def measurement_listener() -> None:
+        """Listen for new measurements and add sensors if they did not exist."""
+        received_measurement_types = set(measurement_coordinator.data.keys())
+        new_measurement_types = received_measurement_types - current_measurement_types
+        if new_measurement_types:
+            async_add_entities(
```

Comment:
For some reason, ruff doesn't allow me to add a 
```python
current_measurement_types = received_measurement_types
```
in here

Reply:
Not sure what you were trying to do.

Can you push it with commit `-n` so it doesn't get `ruff`ed out

Reply:
I wanted to update the list of measurements, so it doesn't add the same sensor the next time it receives data

Reply:
Do you need a `nonlocal`?

Reply:
Or maybe cleaner to do `current_measurement_types.update(....)`